listenbrainz/webserver/views/stats_api.py

```diff
@@ -84,42 +86,141 @@ def get_artist(user_name):
+            may not be present in all the responses
+
+    :param count: Optional, number of releases to return, Default: :data:`~webserver.views.api.DEFAULT_ITEMS_PER_GET`
+        Max: :data:`~webserver.views.api.MAX_ITEMS_PER_GET`
+    :type count: ``int``
+    :param offset: Optional, number of releases to skip from the beginning, for pagination.
+        Ex. An offset of 5 means the top 5 releases will be skipped, defaults to 0
+    :type offset: ``int``
+    :param range: Optional, time interval for which statistics should be collected, possible values are ``week``,
+        ``month``, ``year``, ``all_time``, defaults to ``all_time``
+    :type range: ``str``
+    :statuscode 200: Successful query, you have data!
+    :statuscode 204: Statistics for the user haven't been calculated, empty response will be returned
+    :statuscode 400: Bad request, check ``response['error']`` for more details
+    :statuscode 404: User not found
+    :resheader Content-Type: *application/json*
+    """
     user = db_user.get_by_mb_id(user_name)
     if user is None:
         raise APINotFound("Cannot find user: %s" % user_name)
 
-    stats = db_stats.get_user_artists(user['id'])
+    stats = db_stats.get_user_releases(user['id'])
     if stats is None:
-        return '', 204
+        raise APINoContent('')
+
+    return jsonify(_process_entity(user_name, entity='release', stats=stats))
+
+
+def _process_entity(user_name, entity, stats):
+    """ Process the statistics data according to query params
+
+        Args:
+            user_name (str): musicbrainz_id of the user
+            entity (str): name of the entity, i.e 'artist', 'release' or 'recording'
+            stats (dict): the dictionary containing statistic data
+
+        Returns:
+            processed_data (dict): A dictionary that contains processed data according
+                to the query params
+    """
+
+    stats_range = request.args.get('range', default='all_time')
+    if not _is_valid_range(stats_range):
+        raise APIBadRequest("Invalid range: {}".format(stats_range))
+
+    offset = _get_non_negative_param('offset', default=0)
+    count = _get_non_negative_param('count', default=DEFAULT_ITEMS_PER_GET)
+
+    plural_entity = entity + 's'
 
     count = min(count, MAX_ITEMS_PER_GET)
     try:
-        total_artist_count = stats['artist'][stats_range]['count']
+        total_entity_count = stats[entity][stats_range]['count']
     except KeyError:
-        return '', 204
+        raise APINoContent('')
 
     count = count + offset
-    artist_list = stats['artist'][stats_range]['artists'][offset:count]
+    entity_list = stats[entity][stats_range][plural_entity][offset:count]
 
-    return jsonify({'payload': {
+    return {'payload': {
         "user_id": user_name,
-        "artists": artist_list,
-        "count": len(artist_list),
-        "total_artist_count": total_artist_count,
+        plural_entity: entity_list,
```

Comment:
Instead of doing this, could we handle this more explicitly in the actual API view, like return  entity_list as a list and then put that it json inside the view. It'll duplicate a little code but it'll make the json more explicit, and we can change the json of each view individually if we need it.